lib/server.js

```diff
@@ -1,26 +1,30 @@
 "use strict";
 
-//Modules required
-var Hapi = require("hapi");
+var Hapi           = require("hapi");
+var Bluebird       = require("bluebird");
+var MongoMapper    = require("genesis").MongoMapper;
 
-//Setting the port for the application
-var port = process.env.PORT || 8080;
+var mapper = new MongoMapper(process.env.MONGODB_URL);
+
+Bluebird.promisifyAll(Hapi);
+
+var port   = process.env.PORT || 8080;
+var server = module.exports = new Hapi.Server();
 
-//Server initialization
-var server = new Hapi.Server();
 server.connection({
 	host : process.env.HOST,
 	port : port
 });
 
-server.register ([
-		{
-			register : require("./plugins/rest")
+server.registerAsync ([
+	{
+		register : require("./plugins/rest"),
+		options  : {
+			mapper : mapper
 		}
-	],
-	function () {
-		server.start(function () {
-			console.log("Server started at: " + server.info.uri);
-		});
 	}
-);
+])
+.then(function () {
+	server.startAsync(function () {
```

Comment:
Can you elaborate on why a function is provided as an argument here? If it is not needed then it should be removed.